nemo/src/program_analysis/type_inference.rs

```diff
@@ -812,69 +796,4 @@ mod test {
         let inferred_types = infer_types(&b_and_c_conflict_decl_resolvable).unwrap().0;
         assert_eq!(inferred_types, expected_types);
     }
-
-    #[test]
-    fn infer_types_a_unresolvable_conflict_with_source_decls_of_b_and_c() {
-        let (
-            (basic_rule, exis_rule, rule_with_constant),
-            (fact1, fact2, fact3),
-            (a, b, c, _r, _s, _t, _q),
-        ) = get_test_rules_and_facts_and_predicates();
-
-        let a_unresolvable_conflict_with_source_decls_of_b_and_c = ChaseProgram::new(
-            None,
-            Default::default(),
-            vec![
-                DataSourceDeclaration::new(
-                    b,
-                    NativeDataSource::DsvFile(DsvFile::csv_file(
-                        "",
-                        [TypeConstraint::AtLeast(PrimitiveType::Integer)]
-                            .into_iter()
-                            .collect(),
-                    )),
-                ),
-                DataSourceDeclaration::new(
-                    c,
-                    NativeDataSource::DsvFile(DsvFile::csv_file(
-                        "",
-                        [TypeConstraint::AtLeast(PrimitiveType::Integer)]
-                            .into_iter()
-                            .collect(),
-                    )),
-                ),
-            ],
-            vec![basic_rule, exis_rule, rule_with_constant],
-            vec![fact1, fact2, fact3],
-            [(a, vec![PrimitiveType::Any])].into_iter().collect(),
-            Default::default(),
-        );
-
-        let inferred_types_res = infer_types(&a_unresolvable_conflict_with_source_decls_of_b_and_c);
-        assert!(inferred_types_res.is_err());
-    }
-
-    #[test]
-    fn infer_types_s_decl_unresolvable_conflict_with_fact_values() {
-        let (
-            (basic_rule, exis_rule, rule_with_constant),
-            (fact1, fact2, fact3),
-            (_a, _b, _c, _r, s, _t, _q),
-        ) = get_test_rules_and_facts_and_predicates();
-
-        let s_decl_unresolvable_conflict_with_fact_values = ChaseProgram::new(
-            None,
-            Default::default(),
-            vec![],
-            vec![basic_rule, exis_rule, rule_with_constant],
-            vec![fact1, fact2, fact3],
-            [(s, vec![PrimitiveType::Any, PrimitiveType::Integer])]
-                .into_iter()
-                .collect(),
-            Default::default(),
-        );
-
-        let inferred_types_res = infer_types(&s_decl_unresolvable_conflict_with_fact_values);
-        assert!(inferred_types_res.is_err());
-    }
```

Comment:
Why have these tests been removed?

Reply:
I didn't mean to remove these, I guess they slipped through when I was resolving merge conflicts during the rebase. I'll add them again.